src/planets/Mercury.ts

```diff
@@ -5,10 +5,13 @@ import {DIAMETER_MERCURY} from '../constants/diameters';
 import {EclipticSphericalCoordinates} from '../coordinates/types/CoordinateTypes';
 import {normalizeAngle} from '../utils/angleCalc';
 import {getApparentMagnitudeMercury} from '../utils/magnitudeCalc';
+import TimeOfInterest from '../time/TimeOfInterest';
 import Planet from './Planet';
 
 export default class Mercury extends Planet {
-    protected name = 'mercury';
+    constructor(toi?: TimeOfInterest, protected useVsop87Short: boolean = false) {
```

Comment:
If `protected` is added in the constructor arg, it will shadow the field having the same name in the `Planet` class, where it is already `protected`.  This modifier would better be removed here and kept in the super-class.

I'm also not sure that the default value `= false` should be kept on both levels, one should be enough.